src/fetch/ipfs.py

```diff
@@ -49,3 +56,55 @@ def get_content(self, max_retries: int = 3) -> Optional[Any]:
             except requests.exceptions.ReadTimeout:
                 attempts += 1
         return None
+
+    @classmethod
+    async def fetch_many(
+        cls, missing_rows: list[dict[str, str]], max_retries: int = 3
+    ) -> tuple[list[FoundContent], list[NotFoundContent]]:
```

Comment:
My assumption is that you wanted to speed up the program by fetching multiple rows concurrently. I have no experience with asynchronous operations in python but I'm pretty sure the equivalent code in rust wouldn't do that.

Looks like in order to have true concurrency you would have to:
1. collect the async requests in a list
2. await all of them concurrently with `await asyncio.gather(requests)`

[stackoverflow reference](https://stackoverflow.com/a/47172471)

Reply:
So I definitely agree that this does not look like it does what we would expect, but I was following [this article](https://www.twilio.com/blog/asynchronous-http-requests-in-python-with-aiohttp) and I believe the async aspect of this is encapsulated entirely within this clause

```
async with aiohttp.ClientSession() as session:
```

So that we can functionally manipulate the data we get within our "gather" statement.

I have tested this locally and we do indeed experience a substantial performance improvement.

Reply:
I skimmed the article and section `Utilizing asyncio for improved performance` sounds like what you are describing here.
>So 8 seconds compared to 29 seconds is a huge jump in performance, but we can do even better using the tools that asyncio provides. In the original example, we are using await after each individual HTTP request, which isn't quite ideal. It's still faster than the requests example because we are running everything in coroutines, but we can instead run all of these requests "concurrently" as asyncio tasks and then check the results at the end, using asyncio.ensure_future and asyncio.gather.

Reply:
So would you say that this comment is "resolved"?

Reply:
Actually, I am not sure this even matters. The only time we need this performance boost is on the first run before any of the data has been synced. There are ~2300 appData hashes and they are all picked up on the very first run of this service. once this table has been populated with these records, there are only ever very sparsely any new one and the service doesn't really need to be async. It could happen that we need to drop the table and rebuild although this is not expected to be very often. I think its nice to have a slight performance boost, but since it only ever matters for a "genesis" run of the service I am not sure it needs to be over done. I also fine the gather approach to be more complicated to implement with all this internal data manipulation that needs to be made.

Reply:
In that case this comment can be considered "resolved".